main.py

```diff
@@ -244,10 +244,14 @@ def command_unknown(self, update: Update, context: CallbackContext) -> None:
 
     def command_list_devices(self, update: Update, context: CallbackContext) -> None:
         device_groups = self.get_user(update).device_groups
+        device_filter = ' '.join(context.args).lower()
         devices = dict()
         for device_group in device_groups:
             for device in device_group.get_devices().values():
-                devices[device['label']] = device
+                name = device['label']
+                # lower(): Hack because Python doesn't support case-insensitive searches
+                if not device_filter or device_filter in name.lower():
+                    devices[name] = device
```

Comment:
This looks very odd.  Usually `device` is something from being iterated over `devices`.  And now after iterating over devices the collection is updated with the device? WTF?!

Consider renaming any of them to something different, so this line is not that confusing.

Reply:
I don't understand the comment. `device` is a single device, `devices` is a collection of ... device :)

Let me know what I'm missing.

Reply:
OK, it makes sense now! Plz ignore it.